demos/http/http_demo_s3_upload/demo_config.h

```diff
@@ -0,0 +1,107 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef DEMO_CONFIG_H_
+#define DEMO_CONFIG_H_
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Logging config definition and header files inclusion are required in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "DEMO"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+
+/**
+ * @brief HTTP server port number.
+ *
+ * In general, port 443 is for TLS HTTP connections.
+ */
+#ifndef HTTPS_PORT
+    #define HTTPS_PORT    443
+#endif
+
+/**
+ * @brief Path of the file containing the server's root CA certificate for TLS authentication.
+ *
+ * @note S3 uses the Baltimore Cybertrust root CA certificate. To download this certificate, see
+ * https://baltimore-cybertrust-root.chain-demos.digicert.com/info/index.html
+ *
+ * #define ROOT_CA_CERT_PATH               "...insert here..."
+ */
+
+
+/**
+ * @brief Presigned GET URL generated by python script in common/src/presigned_urls_gen.py
+ *
+ * @note This script requires AWS CLI to be configured. For instructions, see
+ * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+ *
+ * Run this script and paste the output value IOT_DEMO_HTTPS_PRESIGNED_GET_URL into
+ * S3_PRESIGNED_URL below.
+ */
+#define S3_PRESIGNED_GET_URL    ""
+
+
+/**
+ * @brief Presigned PUT URL generated by python script in common/src/presigned_urls_gen.py
+ *
+ * @note This script requires AWS CLI to be configured. For instructions, see
+ * https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+ *
+ * Run this script and paste the output value IOT_DEMO_HTTPS_PRESIGNED_PUT_URL into
+ * S3_PRESIGNED_URL below.
+ */
+#define S3_PRESIGNED_PUT_URL    ""
```

Comment:
If you intend the user to fill this up, put it inside the comment block, then have a
```
#ifndef S3_PRESIGNED_URL
    #error "whatever"
#endif 
```

Reply:
So these should be in comments then like the ROOT_CA_CERT_PATH

Reply:
I had put empty strings for CI purposes, but intend on removing them once they are configured (unless there's a better workaround?)

Reply:
For see, I see you already catch this as a command line parameter: https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1318/files#diff-224512c4a5413ed757baf615abc71943a8b8dba575a61dddd52e6a1781f3c657R48

And I see you recently put them in `#ifndef` and `#endif`s, in this file; this works for CI. Although, instead of these blank default strings, we should have demo fail the compilation if the presigned URL is not present. That way the customer is alerted in the `#error` to use the python script to generate them.  Putting the presigned URL configs in the comments, CI can still configure on the command line build the definitions.

Reply:
Sorry what I meant was I was trying to avoid build check failures as I already had the `#ifndef/#endif`s with `#error` statements present in `http_demo_s3_upload.c` and didn't want to hit the errors, but I now realize I should probably update `ci.yml` instead.